### DISCUSSION TITLE
Edge CSRF protected error

### DISCUSSION THREAD

**Discussion by misway** [Hebilicious/authjs-nuxt#35] [Q&A] [answered]
Hello,

Testing on localhost is fine, but after deploying to Cloudflare, an error occurs with the JS Sign In functionality,  attempting to log in using Native Link Sign In results in the error message: 
{"url":"/api/auth/signin/google","statusCode":500,"statusMessage":"","message":"CSRF protected","stack":""}. 

I've tried both Google and Facebook login, and the same issue happens with both.

Repo: https://github.com/misway/authjs-test
Cloudflare pages build from this repo: https://authjs-test.pages.dev/

Any suggestions? Thank you.

**Comment by Hebilicious** [comment]
@misway Please provide a reproduction and make sure to use the latest version (0.1.7)

**Reply by misway** [reply]
I have added the reproduction link already. Thank you.

**Reply by Hebilicious** [reply, marked as answer]
You need to set the `baseUrl` to your deployed website URL, otherwise the CSRF protection will prevent the callback from working.

**Reply by misway** [reply]
Thank you, it works!
